tests/utilities/test_app.py

```diff
@@ -161,3 +222,92 @@ def disable_input(self) -> None:
 
     def stop_application_mode(self) -> None:
         pass
+
+
+# > The resolution of the monotonic clock on Windows is usually around 15.6 msec.
+# > The best resolution is 0.5 msec.
+# @link https://docs.python.org/3/library/asyncio-platforms.html:
+ASYNCIO_EVENTS_PROCESSING_REQUIRED_PERIOD = 0.025 if WINDOWS else 0.002
+
+
+async def let_asyncio_process_some_events() -> None:
+    await asyncio.sleep(ASYNCIO_EVENTS_PROCESSING_REQUIRED_PERIOD)
+
+
+def mock_textual_timers(
+    *,
+    ticks_granularity_fps: int = 60,
+) -> ContextManager[MockedTimeMoveClockForward]:
+    single_tick_duration = 1.0 / ticks_granularity_fps
+
+    pending_sleep_events: list[tuple[float, asyncio.Event]] = []
+
+    @contextlib.contextmanager
+    def mock_textual_timers_context_manager():
+        # N.B. `start_time` is not used, but it is useful to have when we set breakpoints there :-)
+        start_time = current_time = monotonic()
+
+        # Our replacement for "textual._timer.Timer._sleep":
+        async def sleep_mock(duration: float) -> None:
+            event = asyncio.Event()
+            target_event_monotonic_time = current_time + duration
+            pending_sleep_events.append((target_event_monotonic_time, event))
+            # Ok, let's wait for this Event
+            # (which can only be "unlocked" by calls to `move_clock_forward()`)
+            await event.wait()
+
+        # Our replacement for "textual._timer.Timer.get_time" and "textual.message.Message._get_time":
+        def get_time_mock() -> float:
+            nonlocal current_time
+
+            # let's make the time advance slightly between 2 consecutive calls of this function,
+            # within the same order of magnitude than 2 consecutive calls to ` timer.monotonic()`:
+            current_time += 1.1e-06
+
+            return current_time
+
+        async def move_clock_forward(*, seconds: float) -> tuple[float, int]:
+            nonlocal current_time, start_time
+
+            ticks_count = ceil(seconds * ticks_granularity_fps)
+            activated_timers_count_total = 0
+            for tick_counter in range(ticks_count):
```

Comment:
naive implementation for the moment: for each simulated second we loop 60 times, and for each tick we check if we have some mocked calls to `asyncio.wait([duration])` to trigger